contracts/BondingCurve/factory/CombinedFactory.sol

```diff
@@ -0,0 +1,109 @@
+pragma solidity ^0.5.4;
+
+import "zos-lib/contracts/Initializable.sol";
+import "zos-lib/contracts/application/App.sol";
+import "../BondingCurve.sol";
+import "../curve/BancorCurveLogic.sol";
+import "../dividend/ClaimsToken.sol";
+
+contract CombinedFactory is Initializable {
+
+    string constant BC_DAO_PACKAGE = "bc-dao";
+    string constant BANCOR_CURVE_LOGIC = "BancorCurveLogic";
+    string constant CLAIMS_TOKEN = "ClaimsToken";
+    string constant BONDING_CURVE = "BondingCurve";
+
+    App public app;
+
+    event FundraisingDeployed(
+        address indexed bondingCurve,
+        address indexed claimsToken,
+        address buyCurve,
+        address sellCurve,
+        address indexed sender
+    );
+
+    function initialize (address _appContractAddress) public initializer {
+      app = App(_appContractAddress);
+    }
+
+    function _createBancorCurveLogic(bytes memory _data) public returns (address proxy) {
+        address admin = address(0);
+        return address(app.create(BC_DAO_PACKAGE, BANCOR_CURVE_LOGIC, admin, _data));
+    }
+
+    function _createClaimsToken(bytes memory _data) public returns (address proxy) {
```

Comment:
Is it safe to have all of these public, and to have a single app instance?

Reply:
The App is a singleton due to how zos works, it’s the static address that is kind of a hub + ‘dependency tracker’ for the different versions of bc-dao we publish on a given network. 

If it changed for some reason, we’d be deploying a new factory anyways.

The factory is just a convenience for deployment of a curve or it’s parts really.

The _create methods can be used independently as an on-chain ‘zos create’ if some users desire. I was initially intending them to be internal (as the function name w/ _ suggests), but I think it’s totally fine for users to create instances.

Actually - you make me think of a good point - we should probably link the factory to a specific version of logic contracts rather than the app (which will give us the _latest version_ with the current way I’m doing it. So the instances should be static rather than linked to through the app.